<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title> • coloc</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/simplex/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">coloc</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">3.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/coloc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/vignette.html">UNKNOWN TITLE</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/chr1swallace/coloc">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1></h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/chr1swallace/coloc/blob/master/vignettes/coloc.Rmd"><code>vignettes/coloc.Rmd</code></a></small>
      <div class="hidden name"><code>coloc.Rmd</code></div>

    </div>

    
    
<div id="title-coloc-a-package-for-colocalisation-analyses" class="section level1">
<h1 class="hasAnchor">
<a href="#title-coloc-a-package-for-colocalisation-analyses" class="anchor"></a>+TITLE: Coloc: a package for colocalisation analyses</h1>
</div>
<div id="author-chris-wallace" class="section level1">
<h1 class="hasAnchor">
<a href="#author-chris-wallace" class="anchor"></a>+AUTHOR: Chris Wallace</h1>
</div>
<div id="email-cew54cam-ac-uk" class="section level1">
<h1 class="hasAnchor">
<a href="#email-cew54cam-ac-uk" class="anchor"></a>+EMAIL: <a href="mailto:cew54@cam.ac.uk">cew54@cam.ac.uk</a>
</h1>
</div>
<div id="date-2016-05-19-thu" class="section level1">
<h1 class="hasAnchor">
<a href="#date-2016-05-19-thu" class="anchor"></a>+DATE: 2016-05-19 Thu</h1>
</div>
<div id="description" class="section level1">
<h1 class="hasAnchor">
<a href="#description" class="anchor"></a>+DESCRIPTION:</h1>
</div>
<div id="keywords" class="section level1">
<h1 class="hasAnchor">
<a href="#keywords" class="anchor"></a>+KEYWORDS:</h1>
</div>
<div id="language-en" class="section level1">
<h1 class="hasAnchor">
<a href="#language-en" class="anchor"></a>+LANGUAGE: en</h1>
</div>
<div id="options-h3-numt-toct-nil-t-t-t-t--t-ft-t-t-options-text-latext-skipnil-dnot-logbook-todot-prinil-tagst" class="section level1">
<h1 class="hasAnchor">
<a href="#options-h3-numt-toct-nil-t-t-t-t--t-ft-t-t-options-text-latext-skipnil-dnot-logbook-todot-prinil-tagst" class="anchor"></a>+OPTIONS: H:3 num:t toc:t :nil @:t ::t |:t ^:t -:t f:t *:t &lt;:t #+OPTIONS: TeX:t LaTeX:t skip:nil d:(not LOGBOOK) todo:t pri:nil tags:t</h1>
</div>
<div id="export_select_tags-export" class="section level1">
<h1 class="hasAnchor">
<a href="#export_select_tags-export" class="anchor"></a>+EXPORT_SELECT_TAGS: export</h1>
</div>
<div id="export_exclude_tags-noexport" class="section level1">
<h1 class="hasAnchor">
<a href="#export_exclude_tags-noexport" class="anchor"></a>+EXPORT_EXCLUDE_TAGS: noexport</h1>
</div>
<div id="link_up" class="section level1">
<h1 class="hasAnchor">
<a href="#link_up" class="anchor"></a>+LINK_UP:</h1>
</div>
<div id="link_home" class="section level1">
<h1 class="hasAnchor">
<a href="#link_home" class="anchor"></a>+LINK_HOME:</h1>
</div>
<div id="xslt" class="section level1">
<h1 class="hasAnchor">
<a href="#xslt" class="anchor"></a>+XSLT:</h1>
</div>
<div id="latex_header" class="section level1">
<h1 class="hasAnchor">
<a href="#latex_header" class="anchor"></a>+latex_header: </h1>
</div>
<div id="latex" class="section level1">
<h1 class="hasAnchor">
<a href="#latex" class="anchor"></a>+latex: %</h1>
</div>
<div id="begin_html" class="section level1">
<h1 class="hasAnchor">
<a href="#begin_html" class="anchor"></a>+begin_html</h1>
<!--
%\VignetteEngine{knitr}
%\VignetteIndexEntry{Colocalisation vignette}
-->
</div>
<div id="end_html" class="section level1">
<h1 class="hasAnchor">
<a href="#end_html" class="anchor"></a>+end_html</h1>
<ul>
<li>A brief outline of colocalisation analysis</li>
</ul>
</div>
<div id="begin_html-1" class="section level1">
<h1 class="hasAnchor">
<a href="#begin_html-1" class="anchor"></a>+begin_html</h1>
<p><font color="grey"> <em>Chris Wallace // <a href="http://chr1swallace.github.io">web</a> // <a href="mailto:cew54%20at%20cam.ac.uk">email</a></em><br></font> #+end_html</p>
<p>The coloc package can be used to perform genetic colocalisation analysis of two potentially related phenotypes, to ask whether they share common genetic causal variant(s) in a given region. There are a few key references which this vignette will not duplicate (see below), but, in brief, two approaches are implemented.</p>
<p>First, the proportional approach uses the fact that for two traits sharing causal variants, regression coefficients for either trait against any set of SNPs in the neighbourhood of those variants must be proportional. This test was first proposed by Plagnol et al. [fn:1] in the context of evaluating whether expression of the gene /RPS26/ mediated the association of type 1 diabetes to a region on chromosome 12q13 as had recently been proposed. The test addressed a common issue in genetics, and meant researchers could avoid the need to squint at parallel manhattan plots to decide whether two traits share causal variants. The function =coloc.test()= in this package evolved from code released by Vincent, but no longer available.</p>
<p>However, choosing <em>which</em> SNPs to use for the test is a problem. The obvious choice is to use those most strongly associated with one or other trait to maximise information, but doing so induces bias in the regression coefficients, which in turn leads to increased likelihood of spuriously rejecting the null of colocalisation, ie a quite substantially increased type 1 error rate [fn:2]. I proposed two alternatives to address this problem, either using a principal component summary of genetic variation in the region to overcome the need to select a small set of test SNPs, implemented in =coloc.pcs()= and associated functions, or to use the ideas of Bayesian model averaging to average p values over SNP selections, generating posterior predictive p values, implemented in =coloc.bma()=.</p>
<p>Proportional testing, however, requires individual level genotype data, which are not always available. Claudia Giambartolomei and Vincent Plagnol have proposed an alternative method, which makes use of Jon Wakefield’s work on determining approximate Bayes Factors from p values [fn:3] to generate a Bayesian colocalisation analysis [fn:4], implemented in the function =coloc.summaries()=. Note that it is possible to use Bayesian analysis for proportional testing too, in determining the posterior distribution of the proportionality constant <span class="math inline">\(\eta\)</span>.</p>
</div>
<div id="caption-suitability-of-functions-in-this-package-for-different-data-structures" class="section level1">
<h1 class="hasAnchor">
<a href="#caption-suitability-of-functions-in-this-package-for-different-data-structures" class="anchor"></a>+CAPTION: Suitability of functions in this package for different data structures</h1>
<p>| Function | Approach | Data required | Dependence | | | | | between datasets | |———————-+————————-+——————–+——————| | coloc.abf | Bayesian | summary statistics | independent | | coloc.abf.datasets | Bayesian | raw genotypes | independent | | coloc.bma | Proportional, averages | raw genotypes | independent | | | over models | | | | coloc.pcs/coloc.test | Proportional, based on | raw genotypes | independent | | | PC summary of genotypes | | |</p>
<ul>
<li>Usage</li>
</ul>
<p>Let’s simulate a small dataset, and compare the three methods.</p>
</div>
<div id="begin_src-r-ravel-echofalse" class="section level1">
<h1 class="hasAnchor">
<a href="#begin_src-r-ravel-echofalse" class="anchor"></a>+begin_src R :ravel echo=FALSE</h1>
<p>library(coloc) setClass(“simdata”, representation(df1=“data.frame”,df2=“data.frame”)) setValidity(“simdata”, function(object) { n &lt;- nrow(<a href="mailto:object@df1">object@df1</a>) if(nrow(<a href="mailto:object@df2">object@df2</a>)!=n) return(“nrow of ‘<span class="citation">@df1</span>’ should equal nrow of ‘<span class="citation">@df2</span>’”) }) setMethod(“show”, signature=“simdata”, function(object) { cat(“pair of simulated datasets, with”,ncol(<a href="mailto:object@df1">object@df1</a>)-1,“SNPs and”,nrow(<a href="mailto:object@df1">object@df1</a>),“samples.”) })</p>
<p>sim.data &lt;- function(nsnps=50,nsamples=200,causals=1:2,nsim=1) { cat(“Generate”,nsim,“small sets of data”) ntotal &lt;- nsnps * nsamples * nsim X1 &lt;- matrix(rbinom(ntotal,1,0.4)+rbinom(ntotal,1,0.4),ncol=nsnps) Y1 &lt;- rnorm(nsamples,rowSums(X1[,causals]),2) X2 &lt;- matrix(rbinom(ntotal,1,0.4)+rbinom(ntotal,1,0.4),ncol=nsnps) Y2 &lt;- rnorm(nsamples,rowSums(X2[,causals]),2) colnames(X1) &lt;- colnames(X2) &lt;- paste(“s”,1:nsnps,sep=“”) df1 &lt;- cbind(Y=Y1,X1) df2 &lt;- cbind(Y=Y2,X2) if(nsim==1) { return(new(“simdata”, df1=as.data.frame(df1), df2=as.data.frame(df2))) } else { index &lt;- split(1:(nsamples * nsim), rep(1:nsim, nsamples)) objects &lt;- lapply(index, function(i) new(“simdata”, df1=as.data.frame(df1[i,]), df2=as.data.frame(df2[i,]))) return(objects) } }</p>
<div id="simulate-some-data-and-load-the-coloc-library" class="section level2">
<h2 class="hasAnchor">
<a href="#simulate-some-data-and-load-the-coloc-library" class="anchor"></a>simulate some data and load the coloc library</h2>
<p>set.seed(46411) data &lt;- sim.data(nsamples=1000,nsim=1) library(coloc) #+end_src</p>
<p>If you are not familiar with the analysis of genetic data in R, I would suggest beginning with the [[<a href="http://www.bioconductor.org/packages/release/bioc/html/snpStats.html" class="uri">http://www.bioconductor.org/packages/release/bioc/html/snpStats.html</a>][snpStats package]], which has functions for reading data in a variety of formats and contains a vignette on data input. Once you have your genetic data loaded as a SnpMatrix, =X=, and phenotype information in a vector =Y=, then conversion to a =data.frame= as used above follows : df &lt;- as.data.frame(cbind(Y=Y, as(X,“numeric”)))</p>
<ul>
<li>An aside: fine mapping under a single causal variant assumption The Approximate Bayes Factor colocalisation analysis described in the next section essentially works by fine mapping each trait under a single causal variant assumption and then integrating over those two posterior distributions to calculate probabilities that those variants are shared. Of course, this means we can look at each trait on its own quite simply, and we can do that with the function =finemap.abf=.</li>
</ul>
<p>First we analyse a single trait from our simulated data to generate summary statistics:</p>
</div>
</div>
<div id="begin_src-r" class="section level1">
<h1 class="hasAnchor">
<a href="#begin_src-r" class="anchor"></a>+begin_src R</h1>
<p>if(requireNamespace(“snpStats”)) { library(snpStats) Y1 &lt;- <a href="mailto:data@df1%24Y">data@df1$Y</a> X1 &lt;- new(“SnpMatrix”,as.matrix(<a href="mailto:data@df1">data@df1</a>[,-1])) p1 &lt;- snpStats::p.value(single.snp.tests(phenotype=Y1, snp.data=X1),df=1) maf &lt;- col.summary(X1)[,“MAF”]</p>
<p>get.beta &lt;- function(…) { tmp &lt;- snpStats::snp.rhs.estimates(…, family=“gaussian”) beta &lt;- sapply(tmp,“[[”,“beta”) varbeta &lt;- sapply(tmp, “[[”, “Var.beta”) return(list(beta=beta,varbeta=varbeta)) } b1 &lt;- get.beta(Y1 ~ 1, snp.data=X1) } else { Y1 &lt;- <a href="mailto:data@df1%24Y">data@df1$Y</a> X1 &lt;- as.matrix(<a href="mailto:data@df1">data@df1</a>[,-1]) tests1 &lt;- lapply(1:ncol(X1), function(i) summary(lm(Y1 ~ X1[,i]))$coefficients[2,]) p1 &lt;- sapply(tests1,“[”,4) maf &lt;- colMeans(X2)/2</p>
<p>get.beta &lt;- function(x) { beta &lt;- sapply(x,“[”,1) varbeta &lt;- sapply(x, “[”, 2)^2 return(list(beta=beta,varbeta=varbeta)) } b1 &lt;- get.beta(tests1) } #+end_src</p>
<p>Then we analyse them:</p>
</div>
<div id="begin_src-r-1" class="section level1">
<h1 class="hasAnchor">
<a href="#begin_src-r-1" class="anchor"></a>+BEGIN_SRC R</h1>
<p>my.res &lt;- finemap.abf(dataset=list(beta=b1<span class="math inline">\(beta, varbeta=b1\)</span>varbeta, N=nrow(X1),sdY=sd(Y1),type=“quant”)) head(my.res) tail(my.res) #+END_SRC</p>
<p>The =SNP.PP= column shows the posterior probability that exactly that SNP is causal. Note the last line in this data.frame does not correspond to a SNP, but to the null model, that no SNP is causal.</p>
<p>Finally, if you do have full genotype data as here, while this is a fast method for fine mapping, it can be sensible to consider multiple causal variant models too. One package that allows you to do this is [[<a href="http://github.com/chr1swallace/GUESSFM" class="uri">http://github.com/chr1swallace/GUESSFM</a>][GUESSFM]], described in[fn:5]</p>
<ul>
<li>(Approximate) Bayes Factor colocalisation analyses</li>
</ul>
<p>** Introduction The idea behind the ABF analysis is that the association of each trait with SNPs in a region may be summarised by a vector of 0s and at most a single 1, with the 1 indicating the causal SNP (so, assuming a single causal SNP for each trait). The posterior probability of each possible configuration can be calculated and so, crucially, can the posterior probabilities that the traits share their configurations. This allows us to estimate the support for the following cases:</p>
<ul>
<li>
<span class="math inline">\(H_0\)</span>: neither trait has a genetic association in the region</li>
<li>
<span class="math inline">\(H_1\)</span>: only trait 1 has a genetic association in the region</li>
<li>
<span class="math inline">\(H_2\)</span>: only trait 2 has a genetic association in the region</li>
<li>
<span class="math inline">\(H_3\)</span>: both traits are associated, but with different causal variants</li>
<li>
<span class="math inline">\(H_4\)</span>: both traits are associated and share a single causal variant</li>
</ul>
<p>** The basic =coloc.abf= function The function =coloc.abf= is ideally suited to the case when only summary data are available, and requires, for each trait, either: - p values for each SNP - each SNP’s minor allele frequency - sample size - ratio of cases:controls (if using a case-control trait) or: - regression coefficients for each SNP - variance of these regression coefficients.</p>
<p>If regression coefficients and their variance are available, please use these, but we can also approximate Bayes Factors from p values and minor allele frequencies, although, note, such approximation can be less accurate when imputed data are used. NB, you can mix and match, depending on the data available from each study.</p>
<p>A wrapper function is avalailable to run all these steps but we will first generate the p values manually to give all the details. We use the snpStats library from Bioconductor to calculate the p values quickly. We will also calculate the coefficients and their standard errors to compare the two approaches (but you should use only one - the coefficients if available, otherwise the p values).</p>
</div>
<div id="begin_src-r-2" class="section level1">
<h1 class="hasAnchor">
<a href="#begin_src-r-2" class="anchor"></a>+begin_src R</h1>
<p>if(requireNamespace(“snpStats”)) { library(snpStats)</p>
<p>Y1 &lt;- <a href="mailto:data@df1%24Y">data@df1$Y</a> Y2 &lt;- <a href="mailto:data@df2%24Y">data@df2$Y</a></p>
<p>X1 &lt;- new(“SnpMatrix”,as.matrix(<a href="mailto:data@df1">data@df1</a>[,-1])) X2 &lt;- new(“SnpMatrix”,as.matrix(<a href="mailto:data@df2">data@df2</a>[,-1]))</p>
<p>p1 &lt;- snpStats::p.value(single.snp.tests(phenotype=Y1, snp.data=X1),df=1) p2 &lt;- snpStats::p.value(single.snp.tests(phenotype=Y2, snp.data=X2),df=1)</p>
<p>maf &lt;- col.summary(X2)[,“MAF”]</p>
<p>get.beta &lt;- function(…) { tmp &lt;- snpStats::snp.rhs.estimates(…, family=“gaussian”) beta &lt;- sapply(tmp,“[[”,“beta”) varbeta &lt;- sapply(tmp, “[[”, “Var.beta”) return(list(beta=beta,varbeta=varbeta)) } b1 &lt;- get.beta(Y1 ~ 1, snp.data=X1) b2 &lt;- get.beta(Y2 ~ 1, snp.data=X2) } else { Y1 &lt;- <a href="mailto:data@df1%24Y">data@df1$Y</a> Y2 &lt;- <a href="mailto:data@df2%24Y">data@df2$Y</a></p>
<p>X1 &lt;- as.matrix(<a href="mailto:data@df1">data@df1</a>[,-1]) X2 &lt;- as.matrix(<a href="mailto:data@df2">data@df2</a>[,-1])</p>
<p>tests1 &lt;- lapply(1:ncol(X1), function(i) summary(lm(Y1 ~ X1[,i]))<span class="math inline">\(coefficients[2,]) tests2 &lt;- lapply(1:ncol(X2), function(i) summary(lm(Y2 ~ X2[,i]))\)</span>coefficients[2,])</p>
<p>p1 &lt;- sapply(tests1,“[”,4) p2 &lt;- sapply(tests2,“[”,4)</p>
<p>maf &lt;- colMeans(X2)/2</p>
<p>get.beta &lt;- function(x) { beta &lt;- sapply(x,“[”,1) varbeta &lt;- sapply(x, “[”, 2)^2 return(list(beta=beta,varbeta=varbeta)) } b1 &lt;- get.beta(tests1) b2 &lt;- get.beta(tests2) } #+end_src</p>
<p>Note that we are using the second dataset in that case to compute the minor allele frequencies. This is unlikely to make any significant difference but one could have used dataset 1 instead. It is now possible to compute the probabilities of interest.</p>
<p>With coefficients, and std. deviation of Y known, we have the most accurate inference. Note that if you have a case-control trait, set type=“cc” and supply s, the proportion of samples in the dataset that are cases (from which sdY can be derived), instead of giving sdY directly. #+begin_src R my.res &lt;- coloc.abf(dataset1=list(beta=b1<span class="math inline">\(beta, varbeta=b1\)</span>varbeta, N=nrow(X1),sdY=sd(Y1),type=“quant”), dataset2=list(beta=b2<span class="math inline">\(beta, varbeta=b2\)</span>varbeta, N=nrow(X2),sdY=sd(Y2),type=“quant”), MAF=maf) print(my.res[[1]]) #+end_src</p>
<p>When std. deviation of Y is unknown, coloc will try and estimate sdY from variance of beta and MAF, but this can be noisy, particularly for small datasets, and will produce a warning:</p>
</div>
<div id="begin_src-r-3" class="section level1">
<h1 class="hasAnchor">
<a href="#begin_src-r-3" class="anchor"></a>+begin_src R</h1>
<p>my.res &lt;- coloc.abf(dataset1=list(beta=b1<span class="math inline">\(beta, varbeta=b1\)</span>varbeta, N=nrow(X1),type=“quant”), dataset2=list(beta=b2<span class="math inline">\(beta, varbeta=b2\)</span>varbeta, N=nrow(X2),type=“quant”), MAF=maf) print(my.res[[1]]) #+end_src</p>
<p>When only p values are available, we can use the form:</p>
</div>
<div id="begin_src-r-4" class="section level1">
<h1 class="hasAnchor">
<a href="#begin_src-r-4" class="anchor"></a>+begin_src R</h1>
<p>my.res &lt;- coloc.abf(dataset1=list(pvalues=p1,N=nrow(X1),type=“quant”), dataset2=list(pvalues=p2,N=nrow(X2),type=“quant”), MAF=maf) print(my.res[[1]]) #+end_src</p>
<p>Here, as we have simulated full genotype data in snpStats format, we can use the wrapper function =coloc.abf.datasets()= to combine all the steps shown above.</p>
</div>
<div id="begin_src-r-5" class="section level1">
<h1 class="hasAnchor">
<a href="#begin_src-r-5" class="anchor"></a>+begin_src R</h1>
<p>ct.abf &lt;- coloc.abf.datasets(<a href="mailto:data@df1">data@df1</a>, <a href="mailto:data@df2">data@df2</a>, response1=“Y”, response2=“Y”, type1=“quant”, type2=“quant”) #+end_src</p>
<ul>
<li>Proportional testing</li>
</ul>
</div>
<div id="principal-components" class="section level1">
<h1 class="hasAnchor">
<a href="#principal-components" class="anchor"></a>** Principal components</h1>
<p>The code below first prepares a principal component object by combining the genotypes in the two dataset, then models the most informative components (the minimum set required to capture 80% of the genetic variation) in each dataset, before finally testing whether there is colocalisation between these models.</p>
</div>
<div id="begin_src-r-ravel-figtrue" class="section level1">
<h1 class="hasAnchor">
<a href="#begin_src-r-ravel-figtrue" class="anchor"></a>+begin_src R :ravel fig=TRUE</h1>
<div id="run-a-coloc-with-pcs" class="section level2">
<h2 class="hasAnchor">
<a href="#run-a-coloc-with-pcs" class="anchor"></a>run a coloc with pcs</h2>
<p>pcs &lt;- pcs.prepare(<a href="mailto:data@df1">data@df1</a>[,-1], <a href="mailto:data@df2">data@df2</a>[,-1]) pcs.1 &lt;- pcs.model(pcs, group=1, <a href="mailto:Y=data@df1">Y=data@df1</a>[,1], threshold=0.8) pcs.2 &lt;- pcs.model(pcs, group=2, <a href="mailto:Y=data@df2">Y=data@df2</a>[,1], threshold=0.8) ct.pcs &lt;- coloc.test(pcs.1,pcs.2) #+end_src</p>
<p>The plot shows the estimated coefficients for each principal component modeled for traits 1 and 2 on the x and y axes, with circles showing the 95% confidence region. The points lie close to the line through the origin, which supports a hypothesis of colocalisation.</p>
<p>A little more information is stored in the =ct.pcs= object:</p>
</div>
</div>
<div id="begin_src-r-6" class="section level1">
<h1 class="hasAnchor">
<a href="#begin_src-r-6" class="anchor"></a>+begin_src R</h1>
<p>ct.pcs str(summary(ct.pcs)) #+end_src</p>
<p>The best estimate for the coefficient of proportionality, <span class="math inline">\(\hat{\eta}\)</span>, is 1.13, and the null hypothesis of colocalisation is not rejected with a chisquare statistic of 5.27 based on 7 degrees of freedom (<span class="math inline">\(n-1\)</span> where the <span class="math inline">\(n\)</span> is the number of components tested, and one degree of freedom was used in estimating <span class="math inline">\(\eta\)</span>), giving a p value of 0.63. The =summary()= method returns a named vector of length 4 containing this information.</p>
<p>If more information is needed about <span class="math inline">\(\eta\)</span>, then this is available if the =bayes= argument is supplied:</p>
</div>
<div id="begin_src-r-7" class="section level1">
<h1 class="hasAnchor">
<a href="#begin_src-r-7" class="anchor"></a>+begin_src R</h1>
<p>ct.pcs.bayes &lt;- coloc.test(pcs.1,pcs.2, bayes=TRUE) ci(ct.pcs.bayes) #+end_src</p>
</div>
<div id="bayesian-model-averaging" class="section level1">
<h1 class="hasAnchor">
<a href="#bayesian-model-averaging" class="anchor"></a>** Bayesian model averaging</h1>
</div>
<div id="this-approach-appears-simpler--there-is-no-need-to-do-any" class="section level1">
<h1 class="hasAnchor">
<a href="#this-approach-appears-simpler--there-is-no-need-to-do-any" class="anchor"></a>This approach appears simpler. There is no need to do any</h1>
</div>
<div id="preparatory-work-you-require-only-a-single-function" class="section level1">
<h1 class="hasAnchor">
<a href="#preparatory-work-you-require-only-a-single-function" class="anchor"></a>preparatory work, you require only a single function:</h1>
</div>
<div id="begin_src-r-ravel-figtrue-1" class="section level1">
<h1 class="hasAnchor">
<a href="#begin_src-r-ravel-figtrue-1" class="anchor"></a>#+begin_src R :ravel fig=TRUE</h1>
</div>
<div id="ct-bma---coloc-bmadatadf1-datadf2" class="section level1">
<h1 class="hasAnchor">
<a href="#ct-bma---coloc-bmadatadf1-datadf2" class="anchor"></a>ct.bma &lt;- coloc.bma(<a href="mailto:data@df1">data@df1</a>, <a href="mailto:data@df2">data@df2</a>,</h1>
</div>
<div id="family1gaussian-family2gaussian" class="section level1">
<h1 class="hasAnchor">
<a href="#family1gaussian-family2gaussian" class="anchor"></a>family1=“gaussian”, family2=“gaussian”,</h1>
</div>
<div id="plot-coefftrue" class="section level1">
<h1 class="hasAnchor">
<a href="#plot-coefftrue" class="anchor"></a>plot.coeff=TRUE)</h1>
</div>
<div id="ct-bma-bayes---coloc-bmadatadf1-datadf2" class="section level1">
<h1 class="hasAnchor">
<a href="#ct-bma-bayes---coloc-bmadatadf1-datadf2" class="anchor"></a>ct.bma.bayes &lt;- coloc.bma(<a href="mailto:data@df1">data@df1</a>, <a href="mailto:data@df2">data@df2</a>,</h1>
</div>
<div id="family1gaussian-family2gaussian-1" class="section level1">
<h1 class="hasAnchor">
<a href="#family1gaussian-family2gaussian-1" class="anchor"></a>family1=“gaussian”, family2=“gaussian”,</h1>
</div>
<div id="bayestrue" class="section level1">
<h1 class="hasAnchor">
<a href="#bayestrue" class="anchor"></a>bayes=TRUE)</h1>
</div>
<div id="end_src" class="section level1">
<h1 class="hasAnchor">
<a href="#end_src" class="anchor"></a>#+end_src</h1>
</div>
<div id="the-family12-parameters-are-used-to-specify-the-model-and-link" class="section level1">
<h1 class="hasAnchor">
<a href="#the-family12-parameters-are-used-to-specify-the-model-and-link" class="anchor"></a>The =family1/2= parameters are used to specify the model and link</h1>
</div>
<div id="function-gaussian-with-identity-link-or-binomial-with" class="section level1">
<h1 class="hasAnchor">
<a href="#function-gaussian-with-identity-link-or-binomial-with" class="anchor"></a>function, =“gaussian”= (with identity link) or =“binomial”= (with</h1>
</div>
<div id="logit-link-are-accepted-values--the-plot-here-shows-all-models" class="section level1">
<h1 class="hasAnchor">
<a href="#logit-link-are-accepted-values--the-plot-here-shows-all-models" class="anchor"></a>logit link) are accepted values. The plot here shows all models</h1>
</div>
<div id="considered-simultaneously-and-does-not-attempt-to-discriminate" class="section level1">
<h1 class="hasAnchor">
<a href="#considered-simultaneously-and-does-not-attempt-to-discriminate" class="anchor"></a>considered simultaneously, and does not attempt to discriminate</h1>
</div>
<div id="between-models-with-strong-and-weak-support--its-rather-confusing" class="section level1">
<h1 class="hasAnchor">
<a href="#between-models-with-strong-and-weak-support--its-rather-confusing" class="anchor"></a>between models with strong and weak support. It’s rather confusing,</h1>
</div>
<div id="and-is-off-by-default-" class="section level1">
<h1 class="hasAnchor">
<a href="#and-is-off-by-default-" class="anchor"></a>and is off by default.</h1>
</div>
<div id="however-coloc-bma-is-doing" class="section level1">
<h1 class="hasAnchor">
<a href="#however-coloc-bma-is-doing" class="anchor"></a>However, =coloc.bma()= is doing</h1>
</div>
<div id="quite-some-work-to-cover-the-model-space-efficiently-and-it-is" class="section level1">
<h1 class="hasAnchor">
<a href="#quite-some-work-to-cover-the-model-space-efficiently-and-it-is" class="anchor"></a>quite some work to cover the model space efficiently, and it is</h1>
</div>
<div id="important-to-understand-how-it-does-this--first-the-r2-trim" class="section level1">
<h1 class="hasAnchor">
<a href="#important-to-understand-how-it-does-this--first-the-r2-trim" class="anchor"></a>important to understand how it does this. First, the =r2.trim=</h1>
</div>
<div id="parameter-is-used-to-tag-the-snps---a-subset-of-snps-are-selected-so" class="section level1">
<h1 class="hasAnchor">
<a href="#parameter-is-used-to-tag-the-snps---a-subset-of-snps-are-selected-so" class="anchor"></a>parameter is used to “tag” the SNPs - a subset of SNPs are selected so</h1>
</div>
<div id="that-no-pair-have-r2-r2-trim--the-default-value-is-0-95-and-the" class="section level1">
<h1 class="hasAnchor">
<a href="#that-no-pair-have-r2-r2-trim--the-default-value-is-0-95-and-the" class="anchor"></a>that no pair have <span class="math inline">\(r^2&gt;\)</span> =r2.trim=. The default value is 0.95 and the</h1>
</div>
<div id="idea-is-that-models-containing-snps-with-very-similar-genotypes" class="section level1">
<h1 class="hasAnchor">
<a href="#idea-is-that-models-containing-snps-with-very-similar-genotypes" class="anchor"></a>idea is that models containing SNPs with very similar genotypes</h1>
</div>
<div id="provide-little-additional-information-so-the-p-value-need-be" class="section level1">
<h1 class="hasAnchor">
<a href="#provide-little-additional-information-so-the-p-value-need-be" class="anchor"></a>provide little additional information, so the <span class="math inline">\(p\)</span> value need be</h1>
</div>
<div id="averaged-over-only-one-of-each-such-group--lower-values-of-r2-trim" class="section level1">
<h1 class="hasAnchor">
<a href="#averaged-over-only-one-of-each-such-group--lower-values-of-r2-trim" class="anchor"></a>averaged over only one of each such group. Lower values of =r2.trim=</h1>
</div>
<div id="will-produce-a-sparser-model-space-and-so-decrease-computation-" class="section level1">
<h1 class="hasAnchor">
<a href="#will-produce-a-sparser-model-space-and-so-decrease-computation-" class="anchor"></a>will produce a sparser model space and so decrease computation.</h1>
</div>
<div id="second-the-thr-parameter-is-used-to-discard-snps-which-are" class="section level1">
<h1 class="hasAnchor">
<a href="#second-the-thr-parameter-is-used-to-discard-snps-which-are" class="anchor"></a>Second, the =thr= parameter is used to discard SNPs which are</h1>
</div>
<div id="uninformative-with-regards-the-phenotype-that-is-if-pp_ij-is-the" class="section level1">
<h1 class="hasAnchor">
<a href="#uninformative-with-regards-the-phenotype-that-is-if-pp_ij-is-the" class="anchor"></a>uninformative with regards the phenotype, that is, if <span class="math inline">\(pp_{ij}\)</span> is the</h1>
</div>
<div id="posterior-probability-of-inclusion-in-single-snp-models-for-snp-i" class="section level1">
<h1 class="hasAnchor">
<a href="#posterior-probability-of-inclusion-in-single-snp-models-for-snp-i" class="anchor"></a>posterior probability of inclusion in single SNP models for SNP <span class="math inline">\(i\)</span>,</h1>
</div>
<div id="trait-j-the-set-of-discarded-snps-is-formed-by-those-for-which" class="section level1">
<h1 class="hasAnchor">
<a href="#trait-j-the-set-of-discarded-snps-is-formed-by-those-for-which" class="anchor"></a>trait <span class="math inline">\(j\)</span>, the set of discarded SNPs is formed by those for which</h1>
</div>
<div id="pp_i1-thr-and-pp_i2-thr--models-containing-only-snps" class="section level1">
<h1 class="hasAnchor">
<a href="#pp_i1-thr-and-pp_i2-thr--models-containing-only-snps" class="anchor"></a><span class="math inline">\(pp_{i1}&lt;\)</span> =thr= and <span class="math inline">\(pp_{i2}&lt;\)</span> =thr=. Models containing <em>only</em> SNPs</h1>
</div>
<div id="from-this-set-will-be-ignored--note-that-models-containing-one-snp" class="section level1">
<h1 class="hasAnchor">
<a href="#from-this-set-will-be-ignored--note-that-models-containing-one-snp" class="anchor"></a>from this set will be ignored. Note that models containing one SNP</h1>
</div>
<div id="from-this-set-and-one-snp-not-in-the-set-will-be-evaluated-" class="section level1">
<h1 class="hasAnchor">
<a href="#from-this-set-and-one-snp-not-in-the-set-will-be-evaluated-" class="anchor"></a>from this set and one SNP /not/ in the set <em>will</em> be evaluated.</h1>
</div>
<div id="finally-you-should-tell-coloc-bma-how-many-snps-should-be" class="section level1">
<h1 class="hasAnchor">
<a href="#finally-you-should-tell-coloc-bma-how-many-snps-should-be" class="anchor"></a>Finally, you should tell =coloc.bma()= how many SNPs should be</h1>
</div>
<div id="included-in-each-model--the-default-is-nsnps2-3-appears-slightly" class="section level1">
<h1 class="hasAnchor">
<a href="#included-in-each-model--the-default-is-nsnps2-3-appears-slightly" class="anchor"></a>included in each model. The default is =nsnps=2=, 3 appears slightly</h1>
</div>
<div id="more-powerful-but-will-generally-require-considerably-more" class="section level1">
<h1 class="hasAnchor">
<a href="#more-powerful-but-will-generally-require-considerably-more" class="anchor"></a>more powerful but will generally require considerably more</h1>
</div>
<div id="computation-whilst-values-of-4-and-above-are-both-unlikely-to" class="section level1">
<h1 class="hasAnchor">
<a href="#computation-whilst-values-of-4-and-above-are-both-unlikely-to" class="anchor"></a>computation, whilst values of 4 and above are both unlikely to</h1>
</div>
<div id="provide-more-information-and-very-unlikely-to-be-computed-in-any" class="section level1">
<h1 class="hasAnchor">
<a href="#provide-more-information-and-very-unlikely-to-be-computed-in-any" class="anchor"></a>provide more information and very unlikely to be computed in any</h1>
</div>
<div id="reasonable-time-for-interactive-work-" class="section level1">
<h1 class="hasAnchor">
<a href="#reasonable-time-for-interactive-work-" class="anchor"></a>reasonable time for interactive work.</h1>
<p>** Using Bayes Factors to compare specific values of <span class="math inline">\(\eta\)</span></p>
<p>It may be that specific values of <span class="math inline">\(\eta\)</span> are of interest. For example, when comparing eQTLs in two tissues, or when comparing risk of two related diseases, the value <span class="math inline">\(\eta=1\)</span> is of particular interest. In proportional testing, we can use Bayes Factors to compare the support for different values of <span class="math inline">\(\eta\)</span>. Eg</p>
</div>
<div id="begin_src-r-8" class="section level1">
<h1 class="hasAnchor">
<a href="#begin_src-r-8" class="anchor"></a>+begin_src R</h1>
<div id="compare-individual-values-of-eta" class="section level2">
<h2 class="hasAnchor">
<a href="#compare-individual-values-of-eta" class="anchor"></a>compare individual values of eta</h2>
<p>ct.pcs &lt;- coloc.test(pcs.1,pcs.2, bayes.factor=c(-1,0,1)) bf(ct.pcs)</p>
</div>
<div id="compare-ranges-of-eta" class="section level2">
<h2 class="hasAnchor">
<a href="#compare-ranges-of-eta" class="anchor"></a>compare ranges of eta</h2>
<p>ct.bma &lt;- coloc.bma(<a href="mailto:data@df1">data@df1</a>, <a href="mailto:data@df2">data@df2</a>, family1=“gaussian”, family2=“gaussian”, bayes.factor=list(c(-0.1,1), c(0.9,1.1))) bf(ct.bma) #+end_src</p>
<ul>
<li>The difference between proportional and ABF approaches</li>
</ul>
<p>So what are the differences between proportional and ABF approaches? Which should you choose?</p>
<p>Well, if you only have p values, then you must use ABF. But be aware that a single causal variant is assumed, and that for accurate inference, the causal variant needs to be included, so either very dense or imputed genotyping data is needed. The ABF approach has another big advantage over proportional testing: being Bayesian, it allows you to evaluate support for each hypothesis, whereas with proportional testing, when the null of colocalisation is not rejected, you cannot be sure whether this reflects true colocalisation or a lack of power. The proportional approach is much less affected by spareness of genotyping data, with power only slightly decreased, but type 1 error rates unaffected.</p>
<p>The behaviour of the two approaches only really differs when there are two or more causal variants. When both are shared, proportional testing has the same type 1 error rate, and whilst ABF tends to still favour <span class="math inline">\(H_4\)</span> (shared variant), it tends to put some more weight on <span class="math inline">\(H_3\)</span> (distinct variants). But when at least one is specific to one trait and at least one is shared, their behaviour differs more substantially. Of course, neither approach was designed with this case in mind, and so there is no “right” answer, but it is instructive to understand their expected behaviour. Proportional testing tends to reject colocalisation, whilst ABF tends to favour sharing. Until the methods are extended to incorporate this specific case, it can be useful to compare the two approaches when complete and dense genotyping data are available. When the results differ, we have tended to identify a combination of shared and distinct causal variants. The ABF approach can still be applied in this case, if p values are available conditioning on the strongest signal, as demonstrated in our paper [fn:4].</p>
<p>You can see more about the ABF approach on [[<a href="http://haldanessieve.org/2013/05/21/our-paper-bayesian-test-for-co-localisation-between-pairs-of-genetic-association-studies-using-summary-statistics/" class="uri">http://haldanessieve.org/2013/05/21/our-paper-bayesian-test-for-co-localisation-between-pairs-of-genetic-association-studies-using-summary-statistics/</a>][this blogpost]].</p>
<ul>
<li>Footnotes</li>
</ul>
<p>[fn:1] <a href="http://www.ncbi.nlm.nih.gov/pubmed/19039033" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/19039033</a></p>
<p>[fn:2] <a href="http://onlinelibrary.wiley.com/doi/10.1002/gepi.21765/abstract" class="uri">http://onlinelibrary.wiley.com/doi/10.1002/gepi.21765/abstract</a></p>
<p>[fn:3] <a href="http://www.ncbi.nlm.nih.gov/pubmed/18642345" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/18642345</a></p>
<p>[fn:4] <a href="http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1004383" class="uri">http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1004383</a></p>
<p>[fn:5] <a href="http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1005272" class="uri">http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1005272</a></p>
</div>
</div>
<div id="local-variables" class="section level1">
<h1 class="hasAnchor">
<a href="#local-variables" class="anchor"></a>Local Variables:</h1>
</div>
<div id="firestarter-org-ravel-md-knitr-dispatch" class="section level1">
<h1 class="hasAnchor">
<a href="#firestarter-org-ravel-md-knitr-dispatch" class="anchor"></a>firestarter: (org-ravel-md-knitr-dispatch)</h1>
</div>
<div id="end" class="section level1">
<h1 class="hasAnchor">
<a href="#end" class="anchor"></a>End:</h1>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#title-coloc-a-package-for-colocalisation-analyses">+TITLE: Coloc: a package for colocalisation analyses</a></li>
      <li><a href="#author-chris-wallace">+AUTHOR: Chris Wallace</a></li>
      <li>
<a href="#email-cew54cam-ac-uk">+EMAIL: </a><a href="mailto:cew54@cam-ac-uk">cew54@cam.ac.uk</a>
</li>
      <li><a href="#date-2016-05-19-thu">+DATE: 2016-05-19 Thu</a></li>
      <li><a href="#description">+DESCRIPTION:</a></li>
      <li><a href="#keywords">+KEYWORDS:</a></li>
      <li><a href="#language-en">+LANGUAGE: en</a></li>
      <li><a href="#options-h3-numt-toct-nil-t-t-t-t--t-ft-t-t-options-text-latext-skipnil-dnot-logbook-todot-prinil-tagst">+OPTIONS: H:3 num:t toc:t :nil @:t ::t |:t ^:t -:t f:t *:t &lt;:t #+OPTIONS: TeX:t LaTeX:t skip:nil d:(not LOGBOOK) todo:t pri:nil tags:t</a></li>
      <li><a href="#export_select_tags-export">+EXPORT_SELECT_TAGS: export</a></li>
      <li><a href="#export_exclude_tags-noexport">+EXPORT_EXCLUDE_TAGS: noexport</a></li>
      <li><a href="#link_up">+LINK_UP:</a></li>
      <li><a href="#link_home">+LINK_HOME:</a></li>
      <li><a href="#xslt">+XSLT:</a></li>
      <li><a href="#latex_header">+latex_header: </a></li>
      <li><a href="#latex">+latex: %</a></li>
      <li><a href="#begin_html">+begin_html</a></li>
      <li><a href="#end_html">+end_html</a></li>
      <li><a href="#begin_html-1">+begin_html</a></li>
      <li><a href="#caption-suitability-of-functions-in-this-package-for-different-data-structures">+CAPTION: Suitability of functions in this package for different data structures</a></li>
      <li>
<a href="#begin_src-r-ravel-echofalse">+begin_src R :ravel echo=FALSE</a><ul class="nav nav-pills nav-stacked">
<li><a href="#simulate-some-data-and-load-the-coloc-library">simulate some data and load the coloc library</a></li>
      </ul>
</li>
      <li><a href="#begin_src-r">+begin_src R</a></li>
      <li><a href="#begin_src-r-1">+BEGIN_SRC R</a></li>
      <li><a href="#begin_src-r-2">+begin_src R</a></li>
      <li><a href="#begin_src-r-3">+begin_src R</a></li>
      <li><a href="#begin_src-r-4">+begin_src R</a></li>
      <li><a href="#begin_src-r-5">+begin_src R</a></li>
      <li><a href="#principal-components">** Principal components</a></li>
      <li>
<a href="#begin_src-r-ravel-figtrue">+begin_src R :ravel fig=TRUE</a><ul class="nav nav-pills nav-stacked">
<li><a href="#run-a-coloc-with-pcs">run a coloc with pcs</a></li>
      </ul>
</li>
      <li><a href="#begin_src-r-6">+begin_src R</a></li>
      <li><a href="#begin_src-r-7">+begin_src R</a></li>
      <li><a href="#bayesian-model-averaging">** Bayesian model averaging</a></li>
      <li><a href="#this-approach-appears-simpler--there-is-no-need-to-do-any">This approach appears simpler. There is no need to do any</a></li>
      <li><a href="#preparatory-work-you-require-only-a-single-function">preparatory work, you require only a single function:</a></li>
      <li><a href="#begin_src-r-ravel-figtrue-1">#+begin_src R :ravel fig=TRUE</a></li>
      <li>
<a href="#ct-bma---coloc-bmadatadf1-datadf2">ct.bma &lt;- coloc.bma(</a><a href="mailto:data@df1">data@df1</a>, <a href="mailto:data@df2">data@df2</a>,</li>
      <li><a href="#family1gaussian-family2gaussian">family1=“gaussian”, family2=“gaussian”,</a></li>
      <li><a href="#plot-coefftrue">plot.coeff=TRUE)</a></li>
      <li>
<a href="#ct-bma-bayes---coloc-bmadatadf1-datadf2">ct.bma.bayes &lt;- coloc.bma(</a><a href="mailto:data@df1">data@df1</a>, <a href="mailto:data@df2">data@df2</a>,</li>
      <li><a href="#family1gaussian-family2gaussian-1">family1=“gaussian”, family2=“gaussian”,</a></li>
      <li><a href="#bayestrue">bayes=TRUE)</a></li>
      <li><a href="#end_src">#+end_src</a></li>
      <li><a href="#the-family12-parameters-are-used-to-specify-the-model-and-link">The =family1/2= parameters are used to specify the model and link</a></li>
      <li><a href="#function-gaussian-with-identity-link-or-binomial-with">function, =“gaussian”= (with identity link) or =“binomial”= (with</a></li>
      <li><a href="#logit-link-are-accepted-values--the-plot-here-shows-all-models">logit link) are accepted values. The plot here shows all models</a></li>
      <li><a href="#considered-simultaneously-and-does-not-attempt-to-discriminate">considered simultaneously, and does not attempt to discriminate</a></li>
      <li><a href="#between-models-with-strong-and-weak-support--its-rather-confusing">between models with strong and weak support. It’s rather confusing,</a></li>
      <li><a href="#and-is-off-by-default-">and is off by default.</a></li>
      <li><a href="#however-coloc-bma-is-doing">However, =coloc.bma()= is doing</a></li>
      <li><a href="#quite-some-work-to-cover-the-model-space-efficiently-and-it-is">quite some work to cover the model space efficiently, and it is</a></li>
      <li><a href="#important-to-understand-how-it-does-this--first-the-r2-trim">important to understand how it does this. First, the =r2.trim=</a></li>
      <li><a href="#parameter-is-used-to-tag-the-snps---a-subset-of-snps-are-selected-so">parameter is used to “tag” the SNPs - a subset of SNPs are selected so</a></li>
      <li><a href="#that-no-pair-have-r2-r2-trim--the-default-value-is-0-95-and-the">that no pair have <span class="math inline">\(r^2&gt;\)</span> =r2.trim=. The default value is 0.95 and the</a></li>
      <li><a href="#idea-is-that-models-containing-snps-with-very-similar-genotypes">idea is that models containing SNPs with very similar genotypes</a></li>
      <li><a href="#provide-little-additional-information-so-the-p-value-need-be">provide little additional information, so the <span class="math inline">\(p\)</span> value need be</a></li>
      <li><a href="#averaged-over-only-one-of-each-such-group--lower-values-of-r2-trim">averaged over only one of each such group. Lower values of =r2.trim=</a></li>
      <li><a href="#will-produce-a-sparser-model-space-and-so-decrease-computation-">will produce a sparser model space and so decrease computation.</a></li>
      <li><a href="#second-the-thr-parameter-is-used-to-discard-snps-which-are">Second, the =thr= parameter is used to discard SNPs which are</a></li>
      <li><a href="#uninformative-with-regards-the-phenotype-that-is-if-pp_ij-is-the">uninformative with regards the phenotype, that is, if <span class="math inline">\(pp_{ij}\)</span> is the</a></li>
      <li><a href="#posterior-probability-of-inclusion-in-single-snp-models-for-snp-i">posterior probability of inclusion in single SNP models for SNP <span class="math inline">\(i\)</span>,</a></li>
      <li><a href="#trait-j-the-set-of-discarded-snps-is-formed-by-those-for-which">trait <span class="math inline">\(j\)</span>, the set of discarded SNPs is formed by those for which</a></li>
      <li><a href="#pp_i1-thr-and-pp_i2-thr--models-containing-only-snps"><span class="math inline">\(pp_{i1}&lt;\)</span> =thr= and <span class="math inline">\(pp_{i2}&lt;\)</span> =thr=. Models containing <em>only</em> SNPs</a></li>
      <li><a href="#from-this-set-will-be-ignored--note-that-models-containing-one-snp">from this set will be ignored. Note that models containing one SNP</a></li>
      <li><a href="#from-this-set-and-one-snp-not-in-the-set-will-be-evaluated-">from this set and one SNP /not/ in the set <em>will</em> be evaluated.</a></li>
      <li><a href="#finally-you-should-tell-coloc-bma-how-many-snps-should-be">Finally, you should tell =coloc.bma()= how many SNPs should be</a></li>
      <li><a href="#included-in-each-model--the-default-is-nsnps2-3-appears-slightly">included in each model. The default is =nsnps=2=, 3 appears slightly</a></li>
      <li><a href="#more-powerful-but-will-generally-require-considerably-more">more powerful but will generally require considerably more</a></li>
      <li><a href="#computation-whilst-values-of-4-and-above-are-both-unlikely-to">computation, whilst values of 4 and above are both unlikely to</a></li>
      <li><a href="#provide-more-information-and-very-unlikely-to-be-computed-in-any">provide more information and very unlikely to be computed in any</a></li>
      <li><a href="#reasonable-time-for-interactive-work-">reasonable time for interactive work.</a></li>
      <li>
<a href="#begin_src-r-8">+begin_src R</a><ul class="nav nav-pills nav-stacked">
<li><a href="#compare-individual-values-of-eta">compare individual values of eta</a></li>
      <li><a href="#compare-ranges-of-eta">compare ranges of eta</a></li>
      </ul>
</li>
      <li><a href="#local-variables">Local Variables:</a></li>
      <li><a href="#firestarter-org-ravel-md-knitr-dispatch">firestarter: (org-ravel-md-knitr-dispatch)</a></li>
      <li><a href="#end">End:</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Chris Wallace, Claudia Giambartolomei.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
