<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Coloc: a package for colocalisation analyses • coloc</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/simplex/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Coloc: a package for colocalisation analyses">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">coloc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">4.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/conditioning_05.html">Coloc: relaxing the single causal variant assumption</a>
    </li>
    <li>
      <a href="../articles/enumeration_03.html">Coloc: enumeration of configurations under a single causal variant assumption</a>
    </li>
    <li>
      <a href="../articles/intro_01.html">Coloc: a package for colocalisation analyses</a>
    </li>
    <li>
      <a href="../articles/proportionality_02.html">Coloc: proportionality testing</a>
    </li>
    <li>
      <a href="../articles/sensitivity_04.html">Coloc: sensitivity to prior values</a>
    </li>
    <li>
      <a href="../articles/simdata.html">UNKNOWN TITLE</a>
    </li>
    <li>
      <a href="../articles/vignette.html">Coloc: a package for colocalisation analyses</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/chr1swallace/coloc">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Coloc: a package for colocalisation analyses</h1>
                        <h4 class="author">Chris Wallace</h4>
            
            <h4 class="date">2019-07-07</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/chr1swallace/coloc/blob/master/vignettes/intro_01.Rmd"><code>vignettes/intro_01.Rmd</code></a></small>
      <div class="hidden name"><code>intro_01.Rmd</code></div>

    </div>

    
    
<p><a id="org63a948f"></a></p>
<div id="a-brief-outline-of-colocalisation-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#a-brief-outline-of-colocalisation-analysis" class="anchor"></a>A brief outline of colocalisation analysis</h1>
<p>The coloc package can be used to perform genetic colocalisation analysis of two potentially related phenotypes, to ask whether they share common genetic causal variant(s) in a given region. There are a few key references which this vignette will not duplicate (see below), but, in brief, two approaches are implemented.</p>
<div id="proportional-testing" class="section level2">
<h2 class="hasAnchor">
<a href="#proportional-testing" class="anchor"></a>Proportional testing</h2>
<p>First, the proportional approach uses the fact that for two traits sharing causal variants, regression coefficients for either trait against any set of SNPs in the neighbourhood of those variants must be proportional. This test was first proposed by Plagnol et al. <sup><a id="fnr.1" class="footref" href="#fn.1">1</a></sup> in the context of evaluating whether expression of the gene <em>RPS26</em> mediated the association of type 1 diabetes to a region on chromosome 12q13 as had recently been proposed. The test addressed a common issue in genetics, and meant researchers could avoid the need to squint at parallel manhattan plots to decide whether two traits share causal variants. The function <code><a href="../reference/coloc.test.html">coloc.test()</a></code> in this package evolved from code released by Vincent, but no longer available.</p>
<p>However, choosing <strong>which</strong> SNPs to use for the test is a problem. The obvious choice is to use those most strongly associated with one or other trait to maximise information, but doing so induces bias in the regression coefficients, which in turn leads to increased likelihood of spuriously rejecting the null of colocalisation, ie a quite substantially increased type 1 error rate <sup><a id="fnr.2" class="footref" href="#fn.2">2</a></sup>. I proposed two alternatives to address this problem, either using a principal component summary of genetic variation in the region to overcome the need to select a small set of test SNPs, implemented in <code>coloc.pcs()</code> and associated functions, or to use the ideas of Bayesian model averaging to average p values over SNP selections, generating posterior predictive p values, implemented in <code><a href="../reference/coloc.bma.html">coloc.bma()</a></code>.</p>
<p>See vignette: <a href="proportionality_02.html">proportionality</a></p>
</div>
<div id="enumeration-under-a-single-causal-variant-assumption" class="section level2">
<h2 class="hasAnchor">
<a href="#enumeration-under-a-single-causal-variant-assumption" class="anchor"></a>Enumeration, under a single causal variant assumption</h2>
<p>Proportional testing, however, requires individual level genotype data, which are not always available. Claudia Giambartolomei and Vincent Plagnol proposed an alternative method, which makes use of Jon Wakefield’s work on determining approximate Bayes Factors from p values <sup><a id="fnr.3" class="footref" href="#fn.3">3</a></sup> to generate a enrumeration colocalisation analysis <sup><a id="fnr.4" class="footref" href="#fn.4">4</a></sup>, implemented in the function <code>coloc.summaries()</code>. By assuming there is at most one causal variant per trait, every possible configuration can be individually enumerated and evaluated, and aggregating over these allows us to gauge the relative support for models which support colocalisation to those that don’t.</p>
<p>You can see more about the enumeration approach on <a href="http://haldanessieve.org/2013/05/21/our-paper-bayesian-test-for-co-localisation-between-pairs-of-genetic-association-studies-using-summary-statistics/">this blogpost</a>.</p>
<p>See vignette: <a href="enumeration_03.html">enumeration</a></p>
</div>
<div id="sensitivity-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#sensitivity-analysis" class="anchor"></a>Sensitivity analysis</h2>
<p>As a Bayesian method, coloc.abf() requires the user to specify prior probabilities of SSNP causality and colocalisation. Post-hoc sensitivity analysis can be used to assess whether results are robust across a range of plausible priors.</p>
<p>See vignette: <a href="sensitivity_04.html">sensitivity</a></p>
</div>
<div id="enumeration-relaxing-the-single-causal-variant-assumption" class="section level2">
<h2 class="hasAnchor">
<a href="#enumeration-relaxing-the-single-causal-variant-assumption" class="anchor"></a>Enumeration, relaxing the single causal variant assumption</h2>
<p>The single variant assumption can be relaxed through conditioning. We have now implemented a conditioning step within coloc, which we hope will increase use of conditioning, and proposed an alternative, masking.</p>
<p>See vignette: <a href="conditioning_05.html">conditioning/masking</a></p>
</div>
</div>
<div id="the-difference-between-proportional-and-enumeration-approaches" class="section level1">
<h1 class="hasAnchor">
<a href="#the-difference-between-proportional-and-enumeration-approaches" class="anchor"></a>The difference between proportional and enumeration approaches</h1>
<p>So what are the differences between proportional and enumeration approaches? Which should you choose?</p>
<p>Well, if you only have p values, then you must use enumeration. But be aware that a single causal variant is assumed, and that for accurate inference, the causal variant needs to be included, so either very dense or imputed genotyping data is needed. The enumeration approach has another big advantage over proportional testing: being Bayesian, it allows you to evaluate support for each hypothesis, whereas with proportional testing, when the null of colocalisation is not rejected, you cannot be sure whether this reflects true colocalisation or a lack of power. The proportional approach is much less affected by spareness of genotyping data, with power only slightly decreased, but type 1 error rates unaffected.</p>
<p>The behaviour of the two approaches only really differs when there are two or more causal variants. When both are shared, proportional testing has the same type 1 error rate, and whilst enumeration tends to still favour <span class="math inline">\(H_4\)</span> (shared variant), it tends to put some more weight on <span class="math inline">\(H_3\)</span> (distinct variants). But when at least one is specific to one trait and at least one is shared, their behaviour differs more substantially. Of course, neither approach was designed with this case in mind, and so there is no “right” answer, but it is instructive to understand their expected behaviour. Proportional testing tends to reject colocalisation, whilst enumeration tends to favour sharing. Until the methods are extended to incorporate this specific case, it can be useful to compare the two approaches when complete and dense genotyping data are available. When the results differ, we have tended to identify a combination of shared and distinct causal variants. The enumeration approach can still be applied in this case, if p values are available conditioning on the strongest signal, as demonstrated in our paper <sup><a id="fnr.4.100" class="footref" href="#fn.4">4</a></sup>.</p>
</div>
<div id="footnotes" class="section level1">
<h1 class="hasAnchor">
<a href="#footnotes" class="anchor"></a>Footnotes</h1>
<p><sup><a id="fn.1" href="#fnr.1">1</a></sup><a href="http://www.ncbi.nlm.nih.gov/pubmed/19039033" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/19039033</a></p>
<p><sup><a id="fn.2" href="#fnr.2">2</a></sup><a href="http://onlinelibrary.wiley.com/doi/10.1002/gepi.21765/abstract" class="uri">http://onlinelibrary.wiley.com/doi/10.1002/gepi.21765/abstract</a></p>
<p><sup><a id="fn.3" href="#fnr.3">3</a></sup><a href="http://www.ncbi.nlm.nih.gov/pubmed/18642345" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/18642345</a></p>
<p><sup><a id="fn.4" href="#fnr.4">4</a></sup><a href="http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1004383" class="uri">http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1004383</a></p>
<p><sup><a id="fn.5" href="#fnr.5">5</a></sup><a href="http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1005272" class="uri">http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1005272</a></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#a-brief-outline-of-colocalisation-analysis">A brief outline of colocalisation analysis</a><ul class="nav nav-pills nav-stacked">
<li><a href="#proportional-testing">Proportional testing</a></li>
      <li><a href="#enumeration-under-a-single-causal-variant-assumption">Enumeration, under a single causal variant assumption</a></li>
      <li><a href="#sensitivity-analysis">Sensitivity analysis</a></li>
      <li><a href="#enumeration-relaxing-the-single-causal-variant-assumption">Enumeration, relaxing the single causal variant assumption</a></li>
      </ul>
</li>
      <li><a href="#the-difference-between-proportional-and-enumeration-approaches">The difference between proportional and enumeration approaches</a></li>
      <li><a href="#footnotes">Footnotes</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Chris Wallace, Claudia Giambartolomei.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
