<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Coloc with multiple signals per trait — coloc.signals • coloc</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/simplex/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Coloc with multiple signals per trait — coloc.signals" />
<meta property="og:description" content="New coloc function, builds on coloc.abf() by allowing for multiple
independent causal variants per trait through conditioning or
masking." />
<meta property="og:image" content="https://chr1swallace.github.io/coloc/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">coloc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">5.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/a01_intro.html">Coloc: a package for colocalisation analyses</a>
    </li>
    <li>
      <a href="../articles/a02_data.html">Coloc: data structures</a>
    </li>
    <li>
      <a href="../articles/a03_enumeration.html">Coloc: under a single causal variant assumption</a>
    </li>
    <li>
      <a href="../articles/a04_sensitivity.html">Coloc: sensitivity to prior values</a>
    </li>
    <li>
      <a href="../articles/a06_SuSiE.html">Coloc: using SuSiE to relax the single causal variant assumption</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/chr1swallace/coloc/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Coloc with multiple signals per trait</h1>
    <small class="dont-index">Source: <a href='https://github.com/chr1swallace/coloc/blob/master/R/split.R'><code>R/split.R</code></a></small>
    <div class="hidden name"><code>coloc.signals.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>New coloc function, builds on coloc.abf() by allowing for multiple
independent causal variants per trait through conditioning or
masking.</p>
    </div>

    <pre class="usage"><span class='fu'>coloc.signals</span><span class='op'>(</span>
  <span class='va'>dataset1</span>,
  <span class='va'>dataset2</span>,
  MAF <span class='op'>=</span> <span class='cn'>NULL</span>,
  LD <span class='op'>=</span> <span class='cn'>NULL</span>,
  method <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"single"</span>, <span class='st'>"cond"</span>, <span class='st'>"mask"</span><span class='op'>)</span>,
  mode <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"iterative"</span>, <span class='st'>"allbutone"</span><span class='op'>)</span>,
  p1 <span class='op'>=</span> <span class='fl'>1e-04</span>,
  p2 <span class='op'>=</span> <span class='fl'>1e-04</span>,
  p12 <span class='op'>=</span> <span class='cn'>NULL</span>,
  maxhits <span class='op'>=</span> <span class='fl'>3</span>,
  r2thr <span class='op'>=</span> <span class='fl'>0.01</span>,
  pthr <span class='op'>=</span> <span class='fl'>1e-06</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>dataset1</th>
      <td><p>a list with specifically named elements defining the dataset
to be analysed. See <code><a href='check_dataset.html'>check_dataset</a></code> for details.</p></td>
    </tr>
    <tr>
      <th>dataset2</th>
      <td><p>as above, for dataset 2</p></td>
    </tr>
    <tr>
      <th>MAF</th>
      <td><p>Common minor allele frequency vector to be used for both dataset1 and dataset2, a shorthand for supplying the same vector as parts of both datasets</p></td>
    </tr>
    <tr>
      <th>LD</th>
      <td><p>required if method="cond". matrix of genotype
<em>correlation</em> (ie r, not r^2) between SNPs. If dataset1 and
dataset2 may have different LD, you can instead add LD=LD1 to
the list of dataset1 and a different LD matrix for dataset2</p></td>
    </tr>
    <tr>
      <th>method</th>
      <td><p>default "" means do no conditioning, should return
similar to coloc.abf.  if method="cond", then use conditioning
to coloc multiple signals.  if method="mask", use masking to
coloc multiple signals. if different datasets need different
methods (eg LD is only available for one of them) you can set
method on a per-dataset basis by adding method="..." to the
list for that dataset.</p></td>
    </tr>
    <tr>
      <th>mode</th>
      <td><p>"iterative" or "allbutone".  Easiest understood with
an example.  Suppose there are 3 signal SNPs detected for
trait 1, A, B, C and only one for trait 2, D.</p><pre>Under "iterative" mode, 3 coloc will be performed:
* trait 1 - trait 2
* trait 1 conditioned on A - trait 2
* trait 1 conditioned on A+B - trait 2

Under "allbutone" mode, they would be
* trait 1 conditioned on B+C - trait 2
* trait 1 conditioned on A+C - trait 2
* trait 1 conditioned on A+B - trait 2

Only iterative mode is supported for method="mask".

The allbutone mode is optimal if the signals are known with
certainty (which they never are), because it allows each
signal to be tested without influence of the others.  When
there is uncertainty, it may make sense to use iterative mode,
because the strongest signals aren't affected by conditioning
incorrectly on weaker secondary and less certain signals.
</pre></td>
    </tr>
    <tr>
      <th>p1</th>
      <td><p>prior probability a SNP is associated with trait 1, default 1e-4</p></td>
    </tr>
    <tr>
      <th>p2</th>
      <td><p>prior probability a SNP is associated with trait 2, default 1e-4</p></td>
    </tr>
    <tr>
      <th>p12</th>
      <td><p>prior probability a SNP is associated with both traits, default 1e-5</p></td>
    </tr>
    <tr>
      <th>maxhits</th>
      <td><p>maximum number of levels to condition/mask</p></td>
    </tr>
    <tr>
      <th>r2thr</th>
      <td><p>if masking, the threshold on r2 should be used to
call two signals independent.  our experience is that this
needs to be set low to avoid double calling the same strong
signal.</p></td>
    </tr>
    <tr>
      <th>pthr</th>
      <td><p>if masking or conditioning, what p value threshold to
call a secondary hit "significant"</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>data.table of coloc results, one row per pair of lead snps
detected in each dataset</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Chris Wallace</p>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Chris Wallace, Claudia Giambartolomei.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


